sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/InjectRuntimeFilter.scala

```diff
@@ -281,6 +289,7 @@ object InjectRuntimeFilter extends Rule[LogicalPlan] with PredicateHelper with J
       case join @ ExtractEquiJoinKeys(joinType, leftKeys, rightKeys, _, _, left, right, hint) =>
         var newLeft = left
         var newRight = right
+        val attributeMap = AttributeMap(join.children.flatMap(getAliasMap))
```

Comment:
Just curious, I might have missed some context here:

How can be Join's children to be Project or Aggregate based on 
```
      case p: Project => getAliasMap(p)
      case a: Aggregate => getAliasMap(a)
  ```?

Based on the test case, I assume this is to identify if there is non-simple expression in, for example, join condition?>

Reply:
Yes. It get the original expression from the children if the condition is `AttributeReference`:
<img width="361" alt="image" src="https://user-images.githubusercontent.com/5399861/193483694-a1de6253-9e27-4555-a383-398f25118682.png">

Reply:
ah I see it better now. Thanks!